api/lds.proto

```diff
@@ -96,50 +126,54 @@ message FilterChain {
   // connection as the remote address.
   google.protobuf.BoolValue use_proxy_proto = 4;
 
-  // See base.Metadata description.
+  // [#not-implemented-hide:] filter chain metadata.
   Metadata metadata = 5;
 
-  // See base.TransportSocket description.
+  // [#not-implemented-hide:] See base.TransportSocket description.
   TransportSocket transport_socket = 6;
 }
 
 message Listener {
-  // The unique name of the listener. If no name is provided, Envoy will generate a
-  // UUID for internal use. The name is used for dynamic listener update and removal
-  // via the LDS APIs.
+  // The unique name by which this listener is known. If no name is provided,
+  // Envoy will allocate an internal UUID for the listener. If the listener is to be dynamically
+  // updated or removed via :ref:`LDS <config_listeners_lds>` a unique name must be provided.
+  // By default, the maximum length of a listener's name is limited to 60 characters. This limit can be
+  // increased by setting the :option:`--max-obj-name-len` command line argument to the desired value.
   string name = 1;
 
   // The address that the listener should listen on.
```

Comment:
this must be unique, too. I think. It becomes important if, for example, a new listener tries to bind an existing address with a different name. In this case, the old listener must be deleted before pushing a new listener.